build_runner_core/lib/src/generate/build_impl.dart

```diff
@@ -365,73 +368,77 @@ class _SingleBuild {
-    var tracker =
-        _performanceTracker.startBuilderAction(input, phase.builderLabel);
+      int phaseNumber, InBuildPhase phase, AssetId input) {
+    var pool = _buildPhasePool[phaseNumber] ??= new Pool(16);
+    return pool.withResource(() async {
+      final builder = phase.builder;
+      final outputsHidden = phase.hideOutput;
+      var tracker =
+          _performanceTracker.startBuilderAction(input, phase.builderLabel);
 
-    var builderOutputs = expectedOutputs(builder, input);
+      var builderOutputs = expectedOutputs(builder, input);
 
-    // Add `builderOutputs` to the primary outputs of the input.
-    var inputNode = _assetGraph.get(input);
-    assert(inputNode != null,
-        'Inputs should be known in the static graph. Missing $input');
-    assert(
-        inputNode.primaryOutputs.containsAll(builderOutputs),
-        'input $input with builder $builder missing primary outputs: \n'
-            'Got ${inputNode.primaryOutputs.join(', ')} which was missing:\n' +
-            builderOutputs
-                .where((id) => !inputNode.primaryOutputs.contains(id))
-                .join(', '));
-
-    var wrappedReader = new SingleStepReader(_reader, _assetGraph, phaseNumber,
-        outputsHidden, input.package, _runLazyPhaseForInput);
-
-    if (!await tracker.track(
-        () => _buildShouldRun(builderOutputs, wrappedReader), 'Setup')) {
-      tracker.stop();
-      return <AssetId>[];
-    }
-
-    await _cleanUpStaleOutputs(builderOutputs);
-    await FailureReporter.clean(phaseNumber, input);
-
-    // We may have read some inputs in the call to `_buildShouldRun`, we want
-    // to remove those.
-    wrappedReader.assetsRead.clear();
-
-    var wrappedWriter = new AssetWriterSpy(_writer);
-    var actionDescription =
-        _actionLoggerName(phase, input, _packageGraph.root.name);
-    var logger = new BuildForInputLogger(new Logger(actionDescription));
-
-    actionsStartedCount++;
-    pendingActions
-        .putIfAbsent(phaseNumber, () => new Set<String>())
-        .add(actionDescription);
-
-    await tracker.track(
-        () => runBuilder(builder, [input], wrappedReader, wrappedWriter,
-                    new PerformanceTrackingResolvers(_resolvers, tracker),
-                    logger: logger, resourceManager: _resourceManager)
-                .catchError((_) {
-              // Errors tracked through the logger
-            }),
-        'Build');
-    actionsCompletedCount++;
-    hungActionsHeartbeat.ping();
-    pendingActions[phaseNumber].remove(actionDescription);
+      // Add `builderOutputs` to the primary outputs of the input.
+      var inputNode = _assetGraph.get(input);
+      assert(inputNode != null,
+          'Inputs should be known in the static graph. Missing $input');
+      assert(
+          inputNode.primaryOutputs.containsAll(builderOutputs),
+          'input $input with builder $builder missing primary outputs: \n'
+              'Got ${inputNode.primaryOutputs
```

Comment:
[nit] Consider picking a new manual string break point to avoid the line break inside of it

Reply:
done